feat: Added the Products page

mnath30 opened the pull request:
Added the product page and product context
Features added:
Created the UI of the product page
Added the product data in db
Created Product context

WIP:
The images are not yet hosted and the same images have been used for testing purpose for now, will change it after the filters are implemented. 
Filters and other functionality is not yet added.

mnath30 committed:
feat: Added the Products page

Added the product page and product context

mnath30 committed:
fix: Removed the unused variables

Commented out the unused code which will be used later

prathmeshjagtap commented on src/Components/Filter/Filter.jsx:
```diff
@@ -0,0 +1,65 @@
+                    <h3 className="padding-sm">Category</h3>
+                    <div className="padding-sm">
+                        <input type="checkbox" id="all categories" name="all categories" checked/>
+                        <label for="all categories">All Categories</label>
```
we use forName instead of for in react

ArjunGTX replied:
Are you sure? Isn't it `htmlFor`?

prathmeshjagtap replied:
> Are you sure? Isn't it `htmlFor`?

yes my bad its htmlFor

mnath30 replied:
Thankyou for your review comments.  Made the changes

prathmeshjagtap commented on src/Components/Filter/Filter.jsx:
```diff
@@ -0,0 +1,65 @@
+                        <input type="checkbox" id="all categories" name="all categories" checked/>
+                        <label for="all categories">All Categories</label>
+                        <br/>
+                        <input type="checkbox" id="category 1" name="category 1"/>
+                        <label for="category 1">Category 1</label>
+                        <br/>
+                        <input type="checkbox" id="category 2" name="category 2"/>
+                        <label for="category 2">Category 2</label>
+                        <br/>
+                        <input type="checkbox" id="category 3" name="category 3"/>
+                        <label for="category 3">Category 3</label>
```
you can make an array of categories and map over it instead  of repeating  code

mnath30 replied:
Have made the suggested changes and created an array of products and then mapped over them

prathmeshjagtap commented on src/backend/db/products.js:
```diff
@@ -7,24 +7,182 @@ import { v4 as uuid } from "uuid";
-    price: "1000",
-    categoryName: "fiction",
-  },
+    id:"a1",
```
maintain  id as  _id:uuid(),  because  if you maintain static id, it causes a lot of problems later while implementing  cart and wishlist functionality and becomes  hard to debug

mnath30 replied:
I have made the changes 
I thought that uuid changes on every refresh so that might cause a problem if the page is refreshed

prathmeshjagtap commented on src/pages/Product/Product.jsx:
```diff
@@ -0,0 +1,34 @@
+
+const Product=()=>{
+    const {data}=useProducts();
+    console.log(data);
```
you can remove the console.log

mnath30 replied:
Removed the console log

prathmeshjagtap commented on src/pages/Product/Product.jsx:
```diff
@@ -0,0 +1,34 @@
+    console.log(data);
+    return(
+        <>
+        {console.log(data)}
```
you can remove console  log

prathmeshjagtap commented on src/pages/Product/Product.jsx:
```diff
@@ -0,0 +1,34 @@
+                {/* Product Cards  */}
+                <div className="flex padding-lg">
+                   
+                    {data.map(ele=><ProductCard key={ele.id} source={ele.source} information={ele.information} brand={ele.brand} itemname={ele.itemname} price={ele.price} rating={ele.rating} isAddedToCart={ele.isAddedToCart} isAddedToWishlist={ele.isAddedToWishlist} isOutOfStock={ele.isOutOfStock}/>)}
```
you can direly pass the whole product  and then destructure it in the Product card

mnath30 replied:
Thankyou for your review comment, made the changes

ArjunGTX commented on src/Components/Filter/Filter.jsx:
```diff
@@ -0,0 +1,65 @@
+                    </div>
+                    <h3 className="padding-sm">Rating</h3>
+                    <div className="padding-sm">
+                        <input type="radio" id="4 stars and above"
```
Using spaces in `id` attribute is a bad practice. use `4-stars-and-above` instead

mnath30 replied:
Thankyou for your review comments.  Made the changes

ArjunGTX commented on src/Components/ProductCard/ProductCard.jsx:
```diff
@@ -0,0 +1,29 @@
+// import { useState } from 'react'
```
Why is this commented?

mnath30 replied:
This will be used later in the code and was not useful in this build so commented it

ArjunGTX commented on src/pages/Product/Product.jsx:
```diff
@@ -0,0 +1,34 @@
+            </aside>
+            {/* Product List */}
+            <div className="main-product">
+                {/* Heading */}
```
Isn't this comment redundant? Cause the coded can be easily identified as a heading without the comment too!

mnath30 replied:
Removed the comment

Mohitkumar33 reviewed (commented):
You can use prettier to format the code

mnath30 committed:
refactor: Refactored and removed console logs

Refactored the following:
1. Created arrays to map over the elements
2. Removed the console.log
3. Passed the product and destructured it in products component
4. Changed the for to htmlFor

mnath30 commented:
> You can use prettier to format the code

Ok will keep that in mind thankyou
